### DISCUSSION TITLE
Duplicate Row and delete one of the rows deleted attachment too.

### DISCUSSION THREAD

**Discussion by immohacker** [Budibase/budibase#10656] [Help] [answered]
Hi,
I get an Problem.
i duplicate rows for easier life. In this rows there are fields with type Attachment.
If i delete one of the rows. The attachment is gone in the first row or vice versa.
what make i wrong?
Because i would duplicate an Row delete the attachment and add a new attachment.
It could be possible, that i wish to hold the attachment an then later delete it or not.
Is which order doesnt matter. If i delete it, the attachment is gone in other row
Here screenshots to understand my confusion text ;-)
Single Dataset
![Auswahl_976](https://github.com/Budibase/budibase/assets/109019584/ba8b9267-bc20-488d-b0ec-b5e19857108f)
I click on Duplicate, Dataset duplicated
![Auswahl_977](https://github.com/Budibase/budibase/assets/109019584/59e40f35-5047-4608-887d-cce4d4ebeab8)
Now i delete one from them, which, the original or the duplicated, doesnt matter.
![Auswahl_978](https://github.com/Budibase/budibase/assets/109019584/6a7028bc-b9c9-46fa-a16b-b2e77114e051)
Now the Attachment png is gone. Only the link survives. The link links in void
![Auswahl_979](https://github.com/Budibase/budibase/assets/109019584/b6ed465a-2082-4992-b61c-ec127828a3da)

At the moment i added in the duplication function fields
This fields are from the attachments and i set them to empty text, so that the attachment would not duplicated.
But that is not that what i wish ;-)

Thank you!
Best regards Bonkersdeluxe

**Comment by sghriyadh** [comment, marked as answer]
As i understand, the saved attachments is not duplicated when you duplicate the row. 

It duplicates the attachment link only . once you delete the link in original or duplicate row the attached file deletes from other row as well.

It seems like a bug (attachments don't duplicate but only link to attachment duplicates)

**Reply by melohagan** [reply]
Hi @sghriyadh Thank you answering, this is spot on! I think this one is fairly subjective as to what the appropriate way to handle it is. 

I wouldn't personally want attachments to be duplicated when duplicating a row as there's no need - you've already uploaded the file. 
I can also see though that duplicating the link is also a bit unexpected when deletions occur. Maybe the best approach is to ignore attachment fields and leave them blank when duplicating the row.

If you have any strong opinions about it yourself, feel free to raise an issue!

**Comment by immohacker** [comment]
Hi 
Thank you for reply,
i do it so since ive got the problem.
I set the attachment fields to null in duplicate function.
So i must upload the attachment again.
it would be nice, if the duplicate function get an checkbox, witch can checked duplicate attachments.
But its nice to have.
Background why:
We duplicate car insurances. Only the company of insurances, insurance number , and date changes.
Not the vehicle registration self.
So we must manually upload/add the vehicle registration again.
This is a little work, download from previous insurance upload to new
To see the insurances history, we dont change the insurances self.

Thank you!
Best regards Bonkersdeluxe

**Comment by immohacker** [comment]
Hi it seems its fixed in Version.  2.7.21. 
I tried to duplicate an entry
delete the original an the attachment stays.
and vice versa.
deleted the duplicate and in the original the attachment still exist.
Thank you very much!
Best regards Bonkersdeluxe
